x/datadeal/keeper/deal.go

```diff
@@ -482,3 +482,30 @@ func (k Keeper) RemoveDataDeliveryVote(ctx sdk.Context, vote *types.DataDelivery
 
 	return nil
 }
+
+func (k Keeper) DeactivateDeal(ctx sdk.Context, msg *types.MsgDeactivateDeal) error {
+	deal, err := k.GetDeal(ctx, msg.DealId)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrDealDeactivate, err.Error())
+	}
+
+	if deal.BuyerAddress != msg.RequesterAddress {
+		return sdkerrors.Wrapf(types.ErrDealDeactivate, "invalid deactivate requester")
+	}
+
+	if deal.Status != types.DEAL_STATUS_ACTIVE {
+		return types.ErrDealNotActive
```

Comment:
Let's wrap an error :)
```suggestion
		return sdkerrors.Wrapf(types.ErrDealDeactivate, types.ErrDealNotActive)
```

Reply:
I think it cannot be wrapped an error. So I'll change like it:
```go
return sdkerrors.Wrapf(types.ErrDealDeactivate, "deal's status is not 'ACTIVE'")
```